param1: 0.2
param2: 0.6
param3: 1.0
param4: 1.0
param5: 1.57
param6: 0.0
param7: 0.0
param8: 0.0
param9: 0.8
